docs/program-setup/runtime/Execution.md

```diff
@@ -1,7 +1,34 @@
+When writing an Mbed OS program, in whichever development tool you use, you may need to:
+
+* Set up a connection to your board so you can flash and debug your program.
+    * You won't usually need a driver for Mbed enabled boards
+    * Mbed CLI has a built-in serial terminal to display output from your application
+* Select or create a build profile.
+* Use build rules to include or exclude files and directories from your build.
+* Use the configuration system to define how your board uses features and the bootloader.
+* Modify or set up a new target.
+
 # Execution
 
+Most embedded systems are dedicated to performing a single duty, although some can be very complex. They are typically constrained in resources and contain only the necessary components to perform that duty. Most embedded systems are powered by a single microcontroller chip, which is typically limited in resources such as memory. Software for an embedded system must be carefully designed to make the best use of these limited resources. Mbed OS in the default full profile is often too large for constrained systems, but has a bare metal profile that is targeted to those systems.
```

Comment:
I made this comment on the previous PR - I'm not sure how the first paragraph relates to the rest of the topic, and why the full/bare metal comparison repeats in both paragraphs.

Reply:
Removed the duplicate comparison. I incluced this information because Evelyne asked me to provide an intro to embedded systems/embedded programming. If not relevant, I can remove it.

Reply:
But still, I don't see how the content of this paragraph relates to the rest of this page.

Reply:
Removed